src/hosts.cpp

```diff
@@ -34,7 +34,8 @@ using namespace bc::config;
 
 // TODO: change to network_address bimap hash table with services and age.
 hosts::hosts(const settings& settings)
-  : buffer_(std::max(settings.host_pool_capacity, 1u)),
+  : settings_(settings),
```

Comment:
Don't store settings here, just the value (host_pool_capacity_). I try to avoid propagating the settings class through lesser classes, making their contract more explicit in the construction.